src/CommandingOfficers/Meridian.java

```diff
@@ -0,0 +1,375 @@
+      destinationType = type;
+    }
+    
+    @Override
+    public GameActionSet getPossibleActions(GameMap map, Path movePath, Unit actor)
+    {
+      XYCoord moveLocation = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
+      if( map.isLocationEmpty(actor, moveLocation) )
+      {
+        return new GameActionSet(new TransformAction(actor, movePath, this), false);
+      }
+      return null;
+    }
+
+    @Override
+    public String name()
+    {
+      return String.format("~%s", destinationType);
+    }
+
+    @Override
+    public GameInputState<?> getInputType(StateData data)
+    {
+      return null;
+    }
+  }
+
+  /** Effectively a wait, but the unit ends up as a different unit at the end of it. */
+  public static class TransformAction implements GameAction
+  {
+    private Transform type;
+    private Path movePath;
+    private XYCoord waitLoc = null;
+    private Unit actor = null;
+
+    public TransformAction(Unit unit, Path path, Transform pType)
+    {
+      actor = unit;
+      movePath = path;
+      if( (null != path) && (path.getPathLength() > 0) )
+      {
+        // Store the destination for later.
+        waitLoc = new XYCoord(path.getEnd().x, path.getEnd().y);
+      }
+      type = pType;
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // WAIT actions consist of
+      //   MOVE
+      GameEventQueue transformEvents = new GameEventQueue();
+
+      // Validate input.
+      boolean isValid = true;
+      isValid &= null != actor && !actor.isTurnOver;
+      isValid &= (null != movePath) && (movePath.getPathLength() > 0);
+      isValid &= (null != gameMap);
+      if( isValid )
+      {
+        XYCoord movePathEnd = new XYCoord(movePath.getEnd().x, movePath.getEnd().y);
+        Location moveLocation = gameMap.getLocation(movePathEnd);
+        isValid &= (null == moveLocation.getResident()) || (actor == moveLocation.getResident());
+      }
+
+      // Generate events.
+      if( isValid )
+      {
+        if (Utils.enqueueMoveEvent(gameMap, actor, movePath, transformEvents))
+        {
+          transformEvents.add(new UnitTransformEvent(actor, type.destinationType, type.co));
+        }
+      }
+      return transformEvents;
+    }
+
+    @Override
+    public XYCoord getMoveLocation()
+    {
+      return waitLoc;
+    }
+
+    @Override
+    public XYCoord getTargetLocation()
+    {
+      return waitLoc;
+    }
+
+    @Override
+    public ActionType getType()
+    {
+      return ActionType.UNIT;
+    }
+
+    @Override
+    public String toString()
+    {
+      return String.format("[Move %s to %s and transform to %s]", actor.toStringWithLocation(), waitLoc, type.destinationType);
+    }
+
+    @Override
+    public UnitActionType getUnitActionType()
+    {
+      return type;
+    }
+  }
+
+  /** Effectively a wait, but the unit ends up as a different unit at the end of it. */
+  public static class UnitTransformEvent implements GameEvent
```

Comment:
Should new action types be put in the common action repository just in case they see future use?

Reply:
It would make sense, but these require a strongly-typed reference back to the original Meridian (since there's no event for tranforming) so that's not viable.

Reply:
Hm. Meridian could be reworked to not require a strongly-typed reference in the `UnitTransformEvent` (and that does seem like an odd requirement anyway), but I see what you mean, so I ain't gonna make you change it this time.

Reply:
I mean, I *guess*, since she can keep a list of what type each unit is, and check to see if it transformed after every move action, but that seems like a really terrible idea.

Reply:
You wouldn't have to check after every action, just when you used her ability. And really, I don't see why you couldn't do it the same way, just instead of having the event call `user.justTransformed.add(unit);`, you could just implement `receiveUnitTransformEvent(Unit unit, UnitModel newType)` and add it to the collection there (assuming `unit.CO == this`, of course).
That removes the need for a Meridian reference in `TransformEvent`, and `TransformActionType`/`TransformAction` only need a reference to Meridian so they can hand it to `TransformEvent`.

Reply:
> you could just implement `receiveUnitTransformEvent(Unit unit, UnitModel newType)`

That's of dubious utility. Since you'll be receiving *after* the event happens, you'd want the old type not the new one.
Also, the idea is to allow Commanders to add new action types with minimal effect on other parts of the codebase. A generic event publishing system would work for that, but I'm not convinced I want to add listening for a transform event to the `GameEventListener` overall.

Like, yes, I *could* add it to the `GameEventListener`, but that defeats the point of the exercise.

Reply:
> the idea is to allow Commanders to add new action types with minimal effect on other parts of the codebase

Sure, but "minimal" might still mean "more than you initially hoped".

> but that defeats the point of the exercise.

I dunno about that. If a new kind of thing is happening, it makes sense to have a new kind of event.

Reply:
Should I make Transform a proper basic action type, then?
It *is* fairly extensible ~~and could be used for your eventual werewolf CO~~

Reply:
I wasn't suggesting to make it a standard action type, though I suppose you could. My natural leaning would be to go in the other direction and make all the other actions implicit, like how we didn't have to specify WAIT/LOAD/JOIN in units' available action types before; the game just knows those are available. I'm not sure why we couldn't do the same for ATTACK (has a weapon) and CAPTURE (is infantry), though RESUPPLY would require something different. Then we could do away with the canonical `UnitActionType` arrays. COs could still add new action types to specific models as in this PR.

~~Also, what? I have no idea what you are talking about.~~

Reply:
Hmm. I think I like LOAD/JOIN being explicit, since we could theoretically add units we don't want doing those things.
I'm not sure I'm a super fan of making things implicit in general, either.

~~Mmmhmm. For sure. Also, that entirely depends on the kind of werewolf it is. My Wesnoth mod made it so when you got a werewolf whose day form was a Mage of Light, it gained the ability to transform to a wolf manually any time it wanted (and gained a darkness aura while transformed so it had full control of its form)~~